README.md

```diff
@@ -50,6 +50,28 @@ have dependencies on other algebras which must be implemented.
     - Two promises are equivalent when they yield equivalent values.
     - Two functions are equivalent if they yield equivalent outputs for equivalent inputs.
 
+## Type signature notation
+
+The type signature notation used in this document is described [here](https://sanctuary.js.org/#types). In summary:
+
+1. New types can be created via type constructors. Type constructors take one or more type arguments. For example, the type "array of strings" is written as `Array String`, and the type constructor `Array` can be defined as `Array :: a -> a`.
```

Comment:
@davidchambers Ok it seems to be the case that "type constructors" wrap their type arguments (which can be variables or not (a v.s. Type)).

On interpreting this: `Array :: Type -> Type`, I was thinking "Array, given a Type, will give you back a type of Type". But I was expecting it to give back an "Array of Type" ... not just Type ... and I couldn't resolve this.

But if "by default", any type returned by a type constructor is wrapped by that type constructor, then it makes sense.

```
> :kind Maybe
Maybe :: * -> *
```

`Maybe` given a Type gives back a `Maybe Type`. `Array`, given a type gives back an `Array Type` not just a `Type`.

Thanks!